homeassistant/components/smarthab/__init__.py

```diff
@@ -26,34 +29,62 @@
 )
 
 
-def setup(hass, config) -> bool:
+async def async_setup(hass, config) -> bool:
     """Set up the SmartHab platform."""
 
+    hass.data.setdefault(DOMAIN, {})
     sh_conf = config.get(DOMAIN)
 
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=sh_conf,
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry):
+    """Set up config entry for SmartHab integration."""
+
     # Assign configuration variables
-    username = sh_conf[CONF_EMAIL]
-    password = sh_conf[CONF_PASSWORD]
+    username = entry.data.get(CONF_EMAIL)
+    password = entry.data.get(CONF_PASSWORD)
```

Comment:
```suggestion
    username = entry.data[CONF_EMAIL]
    password = entry.data[CONF_PASSWORD]
```

These are `vol.Required` so they should always be there